### PR TITLE
Add service need option selector for preschool application

### PR DESCRIPTION
**Pull request by patari**
#### Summary

https://github.com/espoon-voltti/evaka/pull/871 for preschool application.

Espoo:

preschool_application_service_need_option_false

Tampere:

preschool_application_service_need_option_true

### CODE REVIEW

File: frontend/src/citizen-frontend/applications/editor/service-need/ServiceTimeSubSectionPreschool.tsx [hunk 1]
```diff
@@ -108,55 +135,124 @@ export default React.memo(function ServiceTimeSubSectionPreschool({
...
+                  )
+                    ? null
+                    : t.validationErrors.unselectableDate
+                }
+                minDate={minDate}
+                maxDate={maxDate}
+                data-qa="connectedDaycarePreferredStartDate-input"
+                id={labelId}
+                required={true}
+                errorTexts={t.validationErrors}
```
Comment (jaahon) [inline frontend/src/citizen-frontend/applications/editor/service-need/ServiceTimeSubSectionPreschool.tsx:175]: Could this (a separate starting date for preschool daycare) be something to move behind a separate feature flag? I don't think it's directly related to being able to select a service need.

**Review by jaahon — COMMENTED** [review]
I don't know if it's coming in a later PR, but employees don't see the selected service need.

**Comment by patari** [comment]
> I don't know if it's coming in a later PR, but employees don't see the selected service need.

Added!

File: frontend/src/citizen-frontend/applications/editor/service-need/ServiceTimeSubSectionPreschool.tsx (hunk 1 above)
Comment (patari) [reply frontend/src/citizen-frontend/applications/editor/service-need/ServiceTimeSubSectionPreschool.tsx:176]: Yes, created a separate flag for this.

File: frontend/src/citizen-frontend/applications/editor/validations.ts [hunk 2]
```diff
@@ -105,17 +105,31 @@ export const validateApplication = (
           terms
         )
       ),
+      connectedDaycarePreferredStartDate:
+        apiData.type === 'PRESCHOOL' &&
+        featureFlags.preschoolApplication.serviceNeedOption
```
Comment (jaahon) [inline frontend/src/citizen-frontend/applications/editor/validations.ts:110]: The new `connectedDaycarePreferredStartDate` feature flag should be used here. Apart from this everything looks good now!

Comment (patari) [reply frontend/src/citizen-frontend/applications/editor/validations.ts]: Fixed.

**Review by jaahon — APPROVED** [review]

**Pull request merged**